### PR TITLE
Update go

### PR DESCRIPTION
**Pull request by GalRogozinski**
Update go to 1.20 
also updated github actions

### CODE REVIEW

File: .github/workflows/lint.yaml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
       - name: Set up Go
         uses: actions/setup-go@v3
         with:
-          go-version: 1.19
+          go-version: 1.20
```
Comment (moshe-blox) [inline .github/workflows/lint.yaml:19]: try this instead:
```yaml
      - name: Setup Go
        uses: actions/setup-go@v4
        with:
          go-version: "1.20.x"
```

**Review by moshe-blox — CHANGES_REQUESTED** [review]

File: go.mod [hunk 2]
```diff
@@ -1,6 +1,6 @@
 module github.com/ssvlabs/ssv-spec
 
-go 1.19
+go 1.20
```
Comment (nkryuchkov) [inline go.mod:3]: Why 1.20 and not 1.22?

Comment (nkryuchkov) [reply go.mod:3]: Well, I think I can answer myself: because we are still using 1.20 in the implementation

Comment (MatheusFranco99) [reply go.mod:3]: Yes

**Review by MatheusFranco99 — APPROVED** [review]

**Review by moshe-blox — APPROVED** [review]

**Pull request merged**